monkey/monkey_island/cc/app.py

```diff
@@ -109,75 +111,81 @@ def init_app_url_rules(app):
 
 
 class FlaskDIWrapper:
+    class URLAlreadyExistsError(Exception):
+        pass
+
     def __init__(self, api: flask_restful.Api, container: DIContainer):
         self._api = api
         self._container = container
 
-    def add_resource(self, resource: Type[flask_restful.Resource], *urls: str):
+    def add_resource(self, resource: Type[IResource]):
         dependencies = self._container.resolve_dependencies(resource)
-        self._api.add_resource(resource, *urls, resource_class_args=dependencies)
+        FlaskDIWrapper._check_for_duplicate_urls(self._api.app.url_map, resource.urls)
+        self._api.add_resource(resource, *resource.urls, resource_class_args=dependencies)
+
+    @staticmethod
+    def _check_for_duplicate_urls(url_map: Map, urls: Sequence[str]):
+        for url in urls:
+            if FlaskDIWrapper._is_url_added(url_map, url):
+                raise FlaskDIWrapper.URLAlreadyExistsError(
+                    f"URL {url} has already been registered!"
+                )
+
+    @staticmethod
+    def _is_url_added(url_map: Map, url_to_add: str) -> bool:
+        return bool(
+            [
+                registered_url
+                for registered_url in url_map.iter_rules()
+                if str(registered_url).strip("/") == url_to_add.strip("/")
```

Comment:
Flask doesn't allow you to add `api/registration` URL, but it could in the future. I don't see a reason why flask couldn't handle the `api/registration` and just prepend the slash

Reply:
My gut says it would essential violate a standard, but I'm not opposed to "future proofing" this just in case. I'll fix my code.